posts/2022-05-25-smart-contract-feature-example.md

```diff
@@ -0,0 +1,234 @@
+ */
+function isValidOrigination(originationDate) {
+  return Date.parse(originationDate) < Date.now()
+}
+
+/**
+ * True if balance is less than or equal to borrower's credit score multiplied by 1000
+ * @param balance on the loan
+ * @param creditScore of the borrower
+ * @returns {boolean} true if credit score is sufficiently large for the balance provided, false otherwise
+ */
+function isValidLoanAmount(balance, creditScore) {
+  // A very simplistic check, which can easily be replaced by a more complex algorithm
+  const score = creditScore * 1000
+  return score >= balance
+}
+```
+
+The `outputMutation` maps the return value from the AWS Lambda function to a predefined mutation that will result in data being written back to the Uni.  The expected result of executing the validation smart contract is a change to a loan's `validationStatus` field and, only if validation was successful, a change to the loan's [data access controls](https://www.vendia.net/blog/multi-party-data-sharing-with-control#data-access-controls) to allow the Servicer to see the (now-validated) loan information.
+
+```graphql
+mutation ValidationOutputMutation($id: ID!,  $validationStatus: Self_Loan_validationStatusEnum!, $acl: [Vendia_Acl_Input_!]) {
+  update_Loan_async(
+    id: $id, 
+    input: {
+      validationStatus: $validationStatus
+    },
+    aclInput: {
+        acl: $acl
+    }
+  ) {
+    error
+  }
+}
+```
+
+
+For more information on how to create a data validation smart contract, please refer to the step-by-step guide in the [feature example](https://github.com/vendia/examples/tree/main/features/share/smart-contracts).
+
+### Invoking a Smart Contract
+
+The data validation smart contract can be invoked programmatically or manually using the Vendia Share web application's Smart Contract view.
+
+The [feature example](https://github.com/vendia/examples/tree/main/features/share/smart-contracts) demonstrates programmatic execution using a set of provided scripts.  For the rest of this post, we'll use the Vendia Share web application to make the invocation and results more visual.
+
+From the **LenderNode** of the [Uni created](https://github.com/vendia/examples/tree/main/features/share/smart-contracts#step-1---create-a-universal-application) in the feature example, click the `Smart Contracts` widget to get a listing of all available smart contracts.
+
+<figure>
+    <img src="https://user-images.githubusercontent.com/85032783/170310807-9cf127d4-cfa0-4ed5-890b-a8500ac5f071.png" />
+    <figcaption ><b>Figure 1</b> - <i>Smart Contract Listing</i></figcaption>
+</figure>
+
+Select the `validation-smart-contract` to view its details, including the ability to manually invoke the smart contract.  Next, click the `Invoke` button in the upper right and provide a valid `loanIdentifier`. The `loanIdentifier` provided is the value that will be sent to the parameterized `inputQuery` defined in the previous section.
```

Comment:
Nitpick, but you you say "click the `Invoke` button in the upper right" but in the modal (which is not the screen to which you are referring to, but is the screen highlighted in the screenshot) the Invoke button is in the lower right. Is it worth adding another screenshot to show the Invoke button in the upper right then this screenshot showing the input into the SC?

Reply:
The rest looks good.